paddle/scripts/docker/README.md

```diff
@@ -1,38 +1,146 @@
+
+   We'd like to give users choices of GPU and no-GPU, because the GPU version image is much larger than then the no-GPU version.
+
+   We'd like to give users choices of AVX and no-AVX, because some cloud providers don't provide AVX-enabled VMs.
+
+## Dockerfile
+
+To realize above goals, we need only one Dockerfile for the development image.  We can put it in the root source directory.
+
+Let us go over our daily development procedure to show how developers can use this file.
+
+1. Check out the source code
 
-1. 用dev image build 我们的源码：
    ```bash
-   docker run -it -p 2022:22 -v $PWD:/paddle paddlepaddle/paddle:dev-0.10.0rc1  /paddle/build.sh
-   ```  
-   注意，这里的 `-v ` 参数把host上的源码目录里的内容映射到了container里的`/paddle` 目录；而container里的 `/paddle/build.sh` 就是源码目录里的 `build.sh`。上述命令调用了本地源码中的 bulid.sh 来build了本地源码，结果在container里的 `/paddle/build` 目录里，也就是本地的源码目录里的 `build` 子目录。
+   git clone https://github.com/PaddlePaddle/Paddle paddle
+   ```
+
+1. Do something
 
-1. 我们希望上述 `build.sh` 脚本在 `build` 子目录里生成一个Dockerfile，使得我们可以运行：
    ```bash
-   docker build -t paddle  ./build
+   cd paddle
+   git checkout -b my_work
+   Edit some files
    ```
-   来生成我们的production image。
-   
-1. 有了这个production image之后，我们可能会希望docker push 到dockerhub.com的我们自己的名下，然后可以用来启动本地或者远程（Kubernetes）jobs：
+
+1. Build/update the development image (if not yet)
 
    ```bash
-   docker tag paddle yiwang/paddle:did-some-change
-   docker push
-   paddlectl run yiwang/paddle:did-some-change /paddle/demo/mnist/train.py
+   docker build -t paddle:dev . # Suppose that the Dockerfile is in the root source directory.
+   ```
+
+1. Build the source code
+
+   ```bash
+   docker run -v $PWD:/paddle -e "GPU=OFF" -e "AVX=ON" -e "TEST=ON" paddle:dev
+   ```
+
+   This command maps the source directory on the host into `/paddle` in the container.
+
+   Please be aware that the default entrypoint of `paddle:dev` is a shell script file `build.sh`, which builds the source code, and outputs to `/paddle/build` in the container, which is actually `$PWD/build` on the host.
+
+   `build.sh` doesn't only build binaries, but also generates a `$PWD/build/Dockerfile` file, which can be used to build the production image.  We will talk about it later.
+
+1. Run on the host (Not recommended)
+
+   If the host computer happens to have all dependent libraries and Python runtimes installed, we can now run/test the built program.  But the recommended way is to running in a production image.
+
+1. Run in the development container
+
+   `build.sh` generates binary files and invokes `make install`.  So we can run the built program within the development container.  This is convenient for developers.
+
+1. Build a production image
+
+    On the host, we can use the `$PWD/build/Dockerfile` to generate a production image.
+
+   ```bash
+   docker build -t paddle --build-arg "BOOK=ON" -f build/Dockerfile .
    ```
 
-   其中 paddlectl 应该是我们自己写的一个脚本，调用kubectl来在Kubernetes机群上启动一个job的。
+1. Run the Paddle Book
```

Comment:
We could use `book` tag as well. Feels like book is more for users trying to learn deep learning than developers.